core/vm/eips.go

```diff
@@ -144,3 +147,115 @@ func enable2929(jt *JumpTable) {
 	jt[SELFDESTRUCT].constantGas = params.SelfdestructGasEIP150
 	jt[SELFDESTRUCT].dynamicGas = gasSelfdestructEIP2929
 }
+
+func enable3074(jt *JumpTable) {
+	jt[AUTH] = &operation{
+		execute:     opAuth,
+		constantGas: params.EcrecoverGas,
+		minStack:    minStack(4, 1),
+		maxStack:    maxStack(4, 1),
+	}
+
+	jt[AUTHCALL] = &operation{
+		execute:     opAuthCall,
+		constantGas: jt[CALL].constantGas,
+		dynamicGas:  jt[CALL].dynamicGas,
+		minStack:    minStack(7, 1),
+		maxStack:    maxStack(7, 1),
+		memorySize:  jt[CALL].memorySize,
+		returns:     true,
+	}
+}
+
+func opAuth(pc *uint64, interpreter *EVMInterpreter, callContext *callCtx) ([]byte, error) {
+	stack := callContext.stack
+	commit, v, r, s := stack.pop(), stack.pop(), stack.pop(), stack.pop()
+
+	// Zero out the current authorized account. Only update it if an address
+	// is successfully recovered from the signature.
+	callContext.authorized = nil
+
+	if v.BitLen() < 8 && crypto.ValidateSignatureValues(byte(v.Uint64()), r.ToBig(), s.ToBig(), true) {
```

Comment:
What is the significance of `v.Bitlen() < 8`?

Reply:
It ensures that `v` can fit into a single byte, which is done in the following statement in the condition. `crypto.ValidateSignatureValues` then ensures it is either `0` or `1`.